README.md

```diff
@@ -2,4 +2,56 @@
 
 [![CircleCI](https://circleci.com/gh/mateusz/sensible-proxy.svg?style=svg)](https://circleci.com/gh/mateusz/sensible-proxy)
 
+Sensible proxy acts as a HTTP/HTTPS proxy that steers all incoming traffic
+to the www sub domain of the requested domain.
 
+## Theory of operation
+
+When using cloud providers it is very common that they will give you a DNS CNAME
+to a load balancer for high availability. Apex domains on the other hand should
+always point to a static IP address. Multiple DNS providers allows you to setup
+an "ALIAS" record that by passes this restriction, but for those without this
+functionality, sensible proxy was created.
+
+Sensible proxy looks at the incoming HTTP(S) connection, finds the requested
+domain and tries to open up a connection to that domain prepended with `www.`.
+If the connection is successful, it will copy TCP traffic between the client and
+the backend acting as a middle man.
```

Comment:
"it will act as the middle man, by copying traffic between the client and the web server"?